libhwmon/wrapadl.cpp

```diff
@@ -188,6 +206,18 @@ int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanp
 	return 0;
 }
 
+int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int* miliwatts)
+{
+	wrap_adlReturn_t rc;
+	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount){
+		return -1;
+	}
+	int* power = new int;
```

Comment:
You're still leaking an int here. You never free it. You do't need to allocate an init here, just use a local variable, and pass a pointer to it.

Reply:
I'd be lost without you haha. I didn't notice that! Guess I'll have to keep de-rusting my c++